### ISSUE TITLE
Script to create a qooxdoo.d.ts and publish to DefinitelyTyped

### ISSUE THREAD

**Issue by johnspackman** [qooxdoo/qooxdoo-compiler#165]
If we publish a qooxdoo.d.ts to https://github.com/DefinitelyTyped/DefinitelyTyped it will be picked up automatically by IDEs (eg Visual Code); there will still be a place for the target output because contribs and developer's own libraries will need to be output, but by publishing to DefinitelyTyped, new users will find that Qooxdoo is recognised by their IDE by default.

The work in creating a qooxdoo.d.ts is already done as part of target output, so all that remains for this issue is to generate qooxdoo.d.ts for just `qx.*` (which might require some kind of include/exclude filter in config.json) and a script to create a PR to https://github.com/DefinitelyTyped/DefinitelyTyped

**Comment by oetiker** [comment]
since the qx compiler can generate this on the fly, there might be an even simpler route ... by putting the project.d.ts file into a suitable location in the project it will get picked up by visual code automatically ... one location where this does work is `source/class/<project>/` but I guess this is somehow configurable ... I just have not yet figured out how :)

**Comment by johnspackman** [comment]
The hybrid approach is the right way to go, because every class has to be compiled in order for type completion to work this is quite an overhead for `qx.*` because of it's shear size and most users will be using a standard release so precompiling is very helpful.

Not sure how we handle versions, I can see that DT support versions so presumably Visual Code will have a way to select which version to use?

**Comment by oetiker** [comment]
I looked to tech details yesterday but did not find anything ... maybe I was just too tired :(

**Comment by cboulanger** [comment]
@oetiker @johnspackman Is this something that should happen before release or should it be postponed?

**Comment by cboulanger** [comment]
@oetiker @johnspackman Thoughts?
